lib/puppet/pops/types/type_factory.rb

```diff
@@ -181,13 +181,7 @@ def self.any
   # @api public
   #
   def self.regexp(pattern = nil)
-    if pattern
-      t = PRegexpType.new(pattern.is_a?(Regexp) ? pattern.inspect[1..-2] : pattern)
-      t.regexp unless pattern.nil? # compile pattern to catch errors
-      t
-    else
-      PRegexpType::DEFAULT
-    end
+    pattern ?  PRegexpType.new(pattern) : PRegexpType::DEFAULT
```

Comment:
Instead of doing this, can the URI pattern set the option inside of the delimiters  - that is possible instead of using a trailing 'x'.
That way there is no chance of trailing options leak into puppet land (at least not via the type).